### DISCUSSION TITLE
Getting started with TWW

### DISCUSSION THREAD

**Discussion by 3nids** [teksi/wastewater#72] [Announcements]
# Prerequisites

Postgresql+postgis has to be installed either locally or on your server. Access configuration is not described here but can be found on the postgresql documentation.

QGIS has to be installed in a minimal 3.34.8 LTR Version, preferably with the osgeo4w installer.

Python modules psychopg2, sqlalchemy and geoalchemy2 have to be installed for INTERLIS IMPORT and export. pum and pirogue will be required for upgrades and database initialisations in the future.

# Setup
**0.  Create the roles on the database cluster :**
https://github.com/teksi/wastewater/blob/main/datamodel/12_0_roles.sql

**1. Install TWW data model using a dump**
Download from the latest Release https://github.com/teksi/wastewater/releases/latest

Or, if you have access, take the last build from the **main** branch from https://github.com/teksi/wastewater/actions/workflows/datamodel-create-dumps.yml

Use the 'Restore' function on your postgres database installation after creating a new database first and select `tww_dev_structure_and_demo_data.backup` (there is not demo data in there currently)
On Windows:
![restore_teksi_database_dump](https://github.com/teksi/wastewater/assets/1265865/d07b9b17-4560-48e2-8ab5-b7521cf39236)

**2. Install plugin from zip**
[teksi_wastewater.2024.0.5.zip](https://github.com/teksi/wastewater/releases/download/2024.0.5/teksi_wastewater.2024.0.5.zip)

You need a Java Runtime Environment to run ili2pg

If you do not want to compromise your existing QGIS installation create a new User profile first in TEKSI and install the plugin version there.
![project_new_user_profile](https://github.com/teksi/wastewater/assets/1265865/f5b82d9f-1ce5-4bfc-a153-e24fca9f51c6)

**3. Download the code** (only if you need separately)
https://github.com/teksi/wastewater/archive/refs/tags/2024.0.zip

**4. Open the project in QGIS**
Use the downloaded projet from the Release or
Download current version from https://github.com/teksi/wastewater/blob/main/project/teksi_wastewater.qgs

Click the button "Download raw file" and save locally
![download qgs raw file](https://github.com/teksi/wastewater/assets/1265865/f2ef9809-ec4d-4094-a632-33c5b61b106a)

Or directly https://raw.githubusercontent.com/teksi/wastewater/main/project/teksi_wastewater.qgs

Create a pg configuration entry with the name: **pg_tww** to match the project files definition

**5. Import from INTERLIS**
Open the project file
![project_open2](https://github.com/teksi/wastewater/assets/1265865/ce2e1d24-15cc-4a0e-94dc-4248458fede0)

Go to the plugin settings and change to Admin Mode - else INTERLIS Import / Export buttons will not be displayed. 
You can also Add Developers Mode and save the logs to have more data available for debugging.
Save settings.
![setting_debugging_mode](https://github.com/teksi/wastewater/assets/1265865/48ece80a-bf62-4abf-9b6b-4a14abbaf565)

Save project file - it is suggested to name it with your QGIS version to distinguish from the provided version - and restart QGIS.
![project_save_as_gis_version](https://github.com/teksi/wastewater/assets/1265865/59ea43ba-478a-4f68-8d78-c0b03cc16f82)

Now the INTERLIS Import / Export buttons should be visible.
![toolbar_teksi_neu](https://github.com/teksi/wastewater/assets/1265865/3807acd4-908b-4139-8045-e86bc1853b65)

Using files in this order (files can be found in the code source)
1. [plugin/teksi_wastewater/tests/data/test-dataset-organisations.xtf ](https://raw.githubusercontent.com/teksi/wastewater/main/plugin/teksi_wastewater/tests/data/test-dataset-organisations.xtf)(use right click - File save as .. to get xtf directly. Needs to be imported first)
2. download the [vsa organisation dataset](https://www.vsa.ch/models/organisation/vsa_organisationen_2020_1.xtf) (needs to bei imported second)
3. download [test-dataset-DSS.xtf](https://raw.githubusercontent.com/teksi/wastewater/main/plugin/teksi_wastewater/tests/data/test-dataset-DSS.xtf) (use right click - File save as .. to get xtf directly. Data - can only succesfully be imported when all organisations referenced are in teksi)

**6. Refresh network views**
Should now be included in the workflow with the latest plugin version.

**7. Run Update Main Cover query manually**
Run the update main cover query manually so symbols on layer vw_tww_wastewater_structure display correctly and objects can be selected - see https://github.com/teksi/wastewater/issues/75
Should now be included in the workflow with the latest plugin version!

# Known issues

* INTERLIS import / export issues have been fixed
* Some 3D geometry errors may happen as they are not fully supported yet. (INTERLIS import is now 2.5D)

LAST EDIT @ponceta 02.04.2025
Fix links

**Comment by sjib** [comment]
I have added some more details in the Setup description (Link to project file, setting for plugin to be able to test INTERLIS Import/Export) to lower the hurdle for users testing.

Testdata is currently under this path (as the plugin is still on a separate branch right now)
[plugin/teksi_wastewater/tests/data](https://github.com/teksi/wastewater/tree/9af5c25dc375a6f0d6d68343a79d0e4c36479f1a/plugin/teksi_wastewater/tests/data)

**Reply by sjib** [reply]
When testing please provide the following imformation - you can copy the template below in the new [issue](https://github.com/teksi/wastewater/issues):

---

```
**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Exact steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots / data**
If applicable, add screenshots or data to help explain your problem.

**Desktop (please complete the following information):**
 - TEKSI plugin version
 - TEKSI datamodel version
 - TEKSI project file version
 - QGIS Version - see the About menu
 - Postgres - you can check with `SELECT version();`
 - Postgis - you can check with `SELECT PostGIS_full_version();`
 - OS
 

**Additional context**
Add any other context about the problem here.
```

---

**Reply by cymed** [reply]
Also when using the last build from https://github.com/teksi/wastewater/actions/workflows/datamodel-create-dumps.yml, check you take the main branch

**Comment by urskaufmann** [comment]
length profil works!!!

**Reply by urskaufmann** [reply]
I can not reproduce no - I get error as before with qgep  https://github.com/QGEP/QGEP/issues/816
but I swear: there was a length profile on my screen 3 days ago!

**Comment by 3nids** [comment]
Project has been updated in the plugin branch to fix #81 
Procedure has been fixed accordingly.

**Comment by 3nids** [comment]
new plugin zip with #81 second fix

**Comment by domi4484** [comment]
New plugin zip with:
- Fix interlis DSS  export (missing ARA and abwasserreinigungsanlage)
- Better interlis error text for missing owner/dataowner/.. references
- Commandline import/export draft

**Reply by sjib** [reply]
@domi4484 If data is imported also with new plugin version interlis export vsa-dss runs thru without errors, but configuration for vsa-dss is not complete as there are most vsadss 2020 classes missing - see https://github.com/teksi/wastewater/issues/98.

**Reply by sjib** [reply]
@domi4484 labels export is still failing - see https://github.com/teksi/wastewater/issues/101

**Comment by urskaufmann** [comment]
Project file has a title (TEKSI Wastewater and General Wastewater Management Plan (GEP))
Because of this title, it's not visible in the QGIS-window-titlebar, which project file is open. We prefer to have an empty title in the projectfile.

**Comment by urskaufmann** [comment]
Creating a child-records if the parent-record is not safed gives a "red" error with a not very understandable errormessage (CRITICAL    Layer vw_access_aid : PostGIS-Fehler beim Attributhinzufügen: ERROR: insert or update on table "structure_part" violates foreign key constraint …).

Is there a possibility to give a better message or even better to tell the user to save first.

Interesting: when inserting a new child-record maintenance event, it works also with this similar red error (I know, it's not a real child because there is a re_table between).

**Reply by urskaufmann** [reply]
Solution: buffered transaction groups, not automatic transaction groups in the project settings

**Reply by ponceta** [reply]
I'll have a check that it is set as default on the demo project.

**Reply by sjib** [reply]
Issue opend for this https://github.com/teksi/wastewater/issues/104

**Comment by urskaufmann** [comment]
Action Digitize Drainage Channel: there is a blue hint in the map who tells me what to do.
Action Digitize: There is no blue hint. This hint is missing

**Reply by sjib** [reply]
- [ ] Add hint: Digitize on the map @ponceta

**Reply by 3nids** [reply]
This has to be done in the action in the project. I would suggest to create a dedicated ticket for this.

**Reply by sjib** [reply]
Issue opened: https://github.com/teksi/wastewater/issues/103

**Comment by ponceta** [comment]
@3nids, Before the first point, on the first iteration, we have to create the roles on the database cluster :

https://github.com/teksi/wastewater/blob/main/datamodel/12_0_roles.sql

Should fix error encountered here :

https://github.com/teksi/wastewater/issues/94

Can you add it to the test steps.

**Comment by domi4484** [comment]
@sjib @cymed @urskaufmann I just uploaded the newest plugin in the top post. This should fix:
- #98 
- Hide logs button when no logs are available. Part of #82 
- #87 
- #79 and #75 

Still open (at least on interlis front):
- #102 
- Labels export #101 and #82 
- #78

**Comment by 3nids** [comment]
datamodel dump updated to [8e0f8f5](https://github.com/teksi/wastewater/commit/8e0f8f54aba1380a0aeeb129b9d403c7e7c17e3a)

**Comment by ponceta** [comment]
Datamodel dump and plugin updated to https://github.com/teksi/wastewater/commit/df6ee39bd1853a6be9a8737abd6284a5ec5b557e 05.03.2024

**Comment by sjib** [comment]
New release candidate published: https://github.com/teksi/wastewater/releases/tag/2024.0.rc2

**Reply by sjib** [reply]
Documentation adapted.

**Comment by TriformMoritz** [comment]
Hello,
This installation and testing process is so complicated that it takes hours to do (it took me some hours and I didi it for the 2015 version allready). 
Testing and installation is, as a consequence not done by many people. If we want people to test, this has to be much easier. I agree the installation of postgresgl (with postis and pgadmin) and Qgis.  And the plugin (via Qgis dépôt).
The rest shouldn't be installed, or shipped with the plugin in Qgis

Problems and suggestions:
- The need to install Java. -> Or it should not be used, or it should come with Qgis.
- The need to install the python packages ->  Or it should not be used, or it should come with Qgis.
- User/ login rules -> Should be via administrator tool of the plugin.
- New database upset -> Should be via administrator tool of the plugin.
- New QGS file (with translation) and entry into the pg_service file-> Should be via administrator tool of the plugin.

So the steps should only be:
1. Install postgres with postgis and set it up (pga_hba.conf file).
2. Install Qgis.
3. Install the plugin.
4. Use the plugin to set up the rest.

So once postgres is setup, the rest is easy and can be done by anyone inexperienced!

I personnaly had a problem installing postgresql 12 and 15 alongside on a windows 10 computer used as a server.
- I have two versions of pgadmin installed.
- The old version would le me acces the postgres 15 server (no 'user info'
- The new version wold not work because the port was used (maybe by the old pgadmin version).

Workaround was using pgadmin (newest version) from my local computer.

Hope this comment is in the right spot. Please tell me if I have to move it.

Moritz

**Comment by sjib** [comment]
New pre-release added https://github.com/teksi/wastewater/releases/tag/2024.0.1

**Reply by sjib** [reply]
@ponceta @cymed @urskaufmann  
Is the additional installation of the linking relationship manager done automatically also from the zip install?

![grafik](https://github.com/user-attachments/assets/efa21505-e0f9-4ad9-b8db-643295e881ca)

I have it installed by zip and then see it:
![grafik](https://github.com/user-attachments/assets/a63633ac-8c3c-41dd-9e7f-7694bb4ad25c)

How to deal with the 'Document Management System'?

**Reply by sjib** [reply]
![grafik](https://github.com/user-attachments/assets/8ebafe71-8c23-49bc-aa9d-2ee3afb441d5)

**Reply by cymed** [reply]
All I did was add the names in the metadata. I lack the experience to know in detail where the problem lies. Maybe @domi4484 can help?

**Reply by domi4484** [reply]
Strange the plugin name is spelled right but for some reason the plugin manager is not finding it. Maybe it doesn't strip the space after `,`. I'll try to find out: #400

![image](https://github.com/user-attachments/assets/002cb2e1-f03b-49c3-8499-827d473a6ea7)
